src/bulk-transfer-mapping.ts

```diff
@@ -0,0 +1,71 @@
+import { log } from '@graphprotocol/graph-ts'
+
+import {
+  BatchPrepared as BatchPreparedEvent,
+  ApproveSRRByCommitmentWithProof as ApproveSRRByCommitmentWithProofEvent,
+  MigrateBatch as MigrateBatchEvent,
+} from '../generated/BulkTransfer/BulkTransfer'
+import { BulkTransfer } from '../generated/schema'
+import { eventUTCMillis, logInvocation, secondsToMillis } from './utils'
+
+export function handleBatchPrepared(event: BatchPreparedEvent): void {
+  logInvocation("handleBatchPrepared", event);
+
+  let merkleRoot = event.params.merkleRoot.toHexString();
+  let batch = BulkTransfer.load(merkleRoot);
+  if (batch != null) {
+    log.info("already received this event for merkleRoot: {}", [
+      event.params.merkleRoot.toString(),
+    ]);
+    return;
+  }
+
+  batch = new BulkTransfer(merkleRoot);
+  batch.txs = [];
+  batch.merkleRoot = event.params.merkleRoot;
+  batch.sender = event.params.sender;
+
+  batch.createdAt = batch.updatedAt = eventUTCMillis(event);
+  batch.save();
+}
+
+export function handleCreateSRRWithProof(event: ApproveSRRByCommitmentWithProofEvent): void {
+  logInvocation("handleCreateSRRWithProof", event);
+
+  let merkleRoot = event.params.merkleRoot.toHexString();
+  let batch = BulkTransfer.load(merkleRoot);
+  if (batch == null) {
+    log.error(
+      "received a CreateSRRWithProof event for an unknown batch. MerkleRoot: {}",
+      [event.params.merkleRoot.toString()]
+    );
+    return;
+  }
+
+  log.info("adding srrApproveHash {}", [event.params.srrApproveHash.toHex()]);
+  // this 3 step assign, push, reassign is necessary here:
+  // (see https://thegraph.com/docs/assemblyscript-api#api-reference):
+  let txs = batch.txs;
+  txs.push(event.params.srrApproveHash);
+  batch.txs = txs;
+  batch.tokenId = event.params.tokenId.toString();
+  batch.updatedAt = eventUTCMillis(event);
+  batch.save();
+}
+
+export function handleMigrateBatch(event: MigrateBatchEvent): void {
```

Comment:
this is not needed. migration is already done so there won't be a BulkTransfer migrate.

Reply:
I think so, removed.